.github/workflows/deploy_release_candidate_pypi.yaml

```diff
@@ -13,11 +13,8 @@ on:
       RC:
         description: Integer RC version for the release (e.g. 3 for RC3)
         required: true
-      PYPI_USER:
-        description: PyPi username to perform the upload with
-        required: true
-      PYPI_PASSWORD:
-        description: PyPi password to perform the upload with
+      PYPI_API_TOKEN:
```

Comment:
feeding username and password result in 403 ... legacy authentication rejected

The finalize release workflow already used API token. Align deploy rc workflow to it